src/modules/Game.class.js

```diff
@@ -20,25 +39,114 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(initialState = Game.deafultInitialState, isForCheck = false) {
+    this.#initialState = initialState.map((row) => [...row]);
+    this.#applyInitialState();
+    this.#isForCheck = isForCheck;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    let isStateChanged = false;
+
+    if (this.#status !== 'playing' && !this.#isForCheck) {
+      return isStateChanged;
+    }
+
+    this.#state.forEach((row) => {
+      const mergedCellsIndexes = new Set();
+      const emptyIndexes = [];
+
+      for (let i = 0; i < row.length; i++) {
+        if (row[i] === 0) {
+          emptyIndexes.push(i);
+          continue;
+        }
+
+        let cellIndex = i;
+
+        if (emptyIndexes.length) {
+          cellIndex = emptyIndexes.shift();
+          emptyIndexes.push(i);
+
+          row[cellIndex] = row[i];
+          row[i] = 0;
+          isStateChanged = true;
+        }
+
+        const previousCellIndex = cellIndex - 1;
+
+        if (
+          previousCellIndex >= 0 &&
+          row[previousCellIndex] === row[cellIndex] &&
+          !mergedCellsIndexes.has(previousCellIndex)
+        ) {
+          mergedCellsIndexes.add(previousCellIndex);
+          row[previousCellIndex] *= 2;
+          row[cellIndex] = 0;
+          emptyIndexes.unshift(cellIndex);
+          this.#score += row[previousCellIndex];
+          isStateChanged = true;
+
+          if (row[previousCellIndex] >= 2048) {
+            this.#status = 'win';
+          }
+        }
+      }
+    });
+
+    if (isStateChanged) {
+      this.#addNumber();
+    }
+
+    if (!this.#isForCheck && !this.#isMovePossible()) {
+      this.#status = 'lose';
+    }
+
+    return isStateChanged;
+  }
+  moveRight() {
+    this.#reverse();
+
+    const result = this.moveLeft();
```

Comment:
The method 'moveRight' relies on 'moveLeft' to perform its logic. If the logic for moving left changes, it could inadvertently affect the behavior of moving right. Be cautious with this interdependency.